### DISCUSSION TITLE
Port forwarding question

### DISCUSSION THREAD

**Discussion by orionbg** [wg-easy/wg-easy#1106] [Q&A]
Hey guys,

I have an Ubuntu 22.04 host that runs wg-easy that host is behind a router and on a 192.168.x.x network
I have configured the router to forward the ports needed for Wireguard and that is working as expected.
What I'm having issues with is the need to open a port through the VPN itself so that when an external (to my router) client connects to it, that port can be used by the client through the VPN and out my router's wan IP... I hope I explained it properly...
Can someone help me out, please?

**Comment by kaaax0815** [comment]
You have to add the ports to the docker compose file itself like in https://github.com/wg-easy/wg-easy/blob/master/docker-compose.yml (see `ports` key)
You have to add rules to your firewall if you use one. depends on your setup as docker bypasses ufw (default on ubuntu) anyway, but could be different depending on your setup.

Then if there are no intermediary firewalls/devices a client should be able to connect to the vpn and use the internet as normal. if you want to be able to reach a client in your local network through the vpn if no firewall is on your server it should work (outbound is normally not blocked)

**Reply by orionbg** [reply]
Thanks for the reply!
I tried that first but the docker compose command failed as it wanted to use that port that I added as the Wireguard port... And I added the port, did not replace the standard one that was already there.
Essentially my setup is as follows:
Ubuntu 22.04 running on a pc (192.168.1.101) that is connected to the router (192.168.1.1)
Wg-easy running in docker on it. (Giving 10.x.x.x IPs)
I have several client devices that connect to Wireguard externally to the router (my android phone, laptop, android tv box, one PC) all of them can connect to WG with no issues and have internet access. I do not need them to communicate with each other but would be very good if they could still see the other devices in their own networks (currently if the VPN is on I loose all local devices like one NAS, Chromecast devices and so on... On android i can make it work but have not yet found a way to do it on windows.)
Anyways, I want to open one port (30100) that will go through Wireguard and out my router (where wg server is) I have set port forwarding on my router but since there are at least three more interfaces to go through (wg Ubuntu PC lan adapter, docker virtual switch, and the client interface) I don't know how to configure it. Been reading online but thing are getting even more tangled then they ere right now. Also that forwarding should work both ways so  other devices can contact the client while the VPN tunnel is open. TCP and UDP protocols.
If any additional info will be beneficial, I'll be happy to provide it.

Best Regards!

**Reply by kaaax0815** [reply]
I dont understand what you want to accomplish. That others on the network can reach wireguard peers, i dont think thats possible (or with really expert knowledge and configs) (for this vpn on the router would be the best, my avm router gives wireguard peers a ip on the correct subnet)

**Reply by orionbg** [reply]
Apologies for the late reply, was very busy and this got sidetracked...
I have a torrent client that needs a port open to communicate with the other peers/seeders
That torrent is running on one of the clients and I need the port to be forwarded through Wireguard so that the torrent client downloads/uploads through the tunnel.
I've put port forwarding on the router (Wireguard side) but the torrent client still says port blocked...

**Reply by kaaax0815** [reply]
@orionbg Maybe look at https://github.com/wg-easy/wg-easy/discussions/1189
